debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/ToAvroMongoDataConverterTest.java

```diff
@@ -52,16 +53,12 @@ record = getFile("complex_objects.json");
 
     @Test
     public void shouldCreateStructWithNestedObject() {
-        for (Entry<String, BsonValue> entry : val.entrySet()) {
-            converter.addFieldSchema(entry, builder);
-        }
+        TreeMap<String, Map<Object, BsonType>> entry = converter.parse(val);
+        converter.buildSchema(entry, builder);
 
-        Schema finalSchema = builder.build();
-        Struct struct = new Struct(finalSchema);
-
-        for (Entry<String, BsonValue> entry : val.entrySet()) {
-            converter.convertRecord(entry, finalSchema, struct);
-        }
+        final Schema finalSchema = builder.build();
+        final Struct struct = new Struct(finalSchema);
+        converter.buildStruct(val, finalSchema, struct);
```

Comment:
@jpechane modifications to test cases while building schema and creating the struct